core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp

```diff
@@ -1402,34 +1482,69 @@ convert_r_u32_to_r_i64(x86::Assembler &a, int32 reg_no_dst, int32 reg_no_src)
 }
 
 /**
- * Encode converting int64 immediate data to int32 register data
+ * Encode converting uint32 immediate data to float register data
  *
  * @param a the assembler to emit the code
- * @param reg_no the no of dst int32 register
- * @param data the src immediate int64 data
+ * @param reg_no the no of dst float register
+ * @param data the src immediate uint32 data
  *
  * @return true if success, false otherwise
  */
 static bool
-convert_imm_i64_to_r_i32(x86::Assembler &a, int32 reg_no, int64 data)
+convert_imm_u32_to_r_f32(x86::Assembler &a, int32 reg_no, uint32 data)
 {
-    return mov_imm_to_r_i32(a, reg_no, (int32)data);
+    /* zero extend by the compiler */
+    mov_imm_to_r_i64(a, REG_I64_FREE_IDX, (int64)data);
+    return mov_r_to_r_f32(a, reg_no, REG_I64_FREE_IDX);
 }
 
 /**
- * Encode converting int64 register data to int32 register data
+ * Encode converting uint32 register data to float register data
  *
  * @param a the assembler to emit the code
- * @param reg_no_dst the no of dst int32 register
- * @param reg_no_src the no of src int64 register
+ * @param reg_no_dst the no of dst uint32 register
+ * @param reg_no_src the no of src float register
  *
  * @return true if success, false otherwise
  */
 static bool
-convert_r_i64_to_r_i32(x86::Assembler &a, int32 reg_no_dst, int32 reg_no_src)
+convert_r_u32_to_r_f32(x86::Assembler &a, int32 reg_no_dst, int32 reg_no_src)
 {
-    a.and_(regs_i64[reg_no_src], 0x00000000FFFFFFFFLL);
-    return mov_r_to_r_i32(a, reg_no_dst, reg_no_src);
+    extend_r32_to_r64(a, REG_I64_FREE_IDX, reg_no_src, false);
+    return mov_r_to_r_f32(a, reg_no_dst, REG_I64_FREE_IDX);
```

Comment:
Should be invalid, the src and dst registers of mov_r_to_r_f32 should be both f64 type. Here REG_I64_FREE_IDX is i64 register no.